Python/Product/PythonTools/PythonToolsPackage.cs

```diff
@@ -248,6 +264,16 @@ protected override Task<object> InitializeToolWindowAsync(Type toolWindowType, i
             return base.CreateToolWindow(ref toolWindowType, id);
         }
 
+        protected override int QueryClose(out bool canClose) {
+            var res = base.QueryClose(out canClose);
+
+            if (canClose) {
+                var pyService = this.GetPythonToolsService();
```

Comment:
Can it return null?

Reply:
The service is created in package initialization, which I presume will have been called if QueryClose is called.